sys/fmt/fmt.c

```diff
@@ -137,6 +137,10 @@ uint32_t scn_u32_dec(const char *str, size_t n)
 
 void print(const char *s, size_t n)
 {
+#ifdef __WITH_AVRLIBC__
+    /* AVR's libc doesn't offer write(), so use fwrite() instead */
+    fwrite(s, n, 1, stdout);
```

Comment:
I thought fwrite abstracts system-call induced half-writes and only returns 0 (instead of 1 for 1 member of size n) on eof or a hard error. No idea if they can happen on stdout on AVR.